state/postgres_types.go

```diff
@@ -21,3 +21,18 @@ type PostgresType struct {
 	EnumValues        []string
 	CompositeAttrs    [][2]string
 }
+
+// XidToXid8 - Converts Xid (32-bit transaction ID) to Xid8 (64-bit FullTransactionId)
+// by calculating and adding an epoch from the current transaction ID
+func XidToXid8(xid Xid, currentXactId Xid8) Xid8 {
```

Comment:
Nice refactoring!

Should we put this in another file though? `state/postgres_types.go` refers to "custom types the collector retrieved", so its not really the same as "type definitions as defined by Postgres".

Maybe `state/util.go` would be a good file?